### ISSUE TITLE
ranking of candidate products

### ISSUE THREAD

**Issue by sparklytopaz** [awslabs/dgl-lifesci#100]
for the pretrained model, how is the ranking of the candidate products done? 
Is there any way I can print it as a probability or confidence score of some kind?
![image](https://user-images.githubusercontent.com/31735343/94779287-521d2280-03e4-11eb-8538-089e048d9e51.png)
I understand that this function ranks the candidate products - but can we convert the tensor to a better ranking evaluation like confidence score from 0 to 1 or probability in percent?

**Comment by mufeili** [comment]
During training, the script uses cross entropy for the predictions corresponding to each reaction as [here](https://github.com/awslabs/dgl-lifesci/blob/master/examples/reaction_prediction/rexgen_direct/candidate_ranking_train.py#L100). That says, you can apply softmax to the candidate products corresponding to a reaction and interpret the output as a categorical distribution.

**Comment by mufeili** [comment]
@sparklytopaz Have you solved the issue?

**Comment by sparklytopaz** [comment]
@mufeili can you please elaborate the process of applying the softmax function?

**Comment by mufeili** [comment]
You can do something like

```python
import torch.nn.functional as F

F.softmax(reaction_pred.reshape(1, -1), dim=1)
```

**Comment by sparklytopaz** [comment]
So this probability will be the probability of the product forming right? Or the probability wrt the ground truth.

**Comment by mufeili** [comment]
> So this probability will be the probability of the product forming right? Or the probability wrt the ground truth.

What's the difference between "the product forming" and "the ground truth"?

**Comment by sparklytopaz** [comment]
What I am trying to ask is can the model be modified to do predictions even if we dont give the product smile

**Comment by mufeili** [comment]
The model already does that. During training, we always include the ground truth while during test we don't need to do that.
